### ISSUE TITLE
[v1] QUploader - missing @closed event

### ISSUE THREAD

**Issue by tomers** [quasarframework/quasar#3480]
There is no way to tell if the user has aborted the upload.

In my use case I need the @closed event because I read the file info after it was added (yes, I am aware it is not that safe and that I should not count on it), and present the user with a QBanner in case the file type is .png instead of .jpg, the more preferred file type from these two supported file types.
In case the user closes the upload component, the banner still shows and I have no way to tell that the uploaded closed in order to hide it.

**Comment by smolinari** [comment]
Hey @tomers 
 
> In case the user closes the upload component

Do you mean the added file? Or the component itself? If the component, how does a user close it? 

Maybe a codepen would be easier to work/ discuss about this?

Scott

**Comment by tomers** [comment]
https://codepen.io/tomers-the-encoder/pen/qvqeaR?editors=1111

You can add files to upload, and should see the banner specifying how many files where added. Once you press the 'x' icon on each file in order to cancel it, there is no way to receive notification that this happened.

**Comment by smolinari** [comment]
Ahhh... very nice. See. Totally understand your use case now. Thanks! 👍 

And, I think this is partially a bug or rather, I found one. Because, if you add more than one file, the message doesn't get updated to two or more files. 

In fact, I fixed this. 

I've also added an `@removed` event, which will do what you are expecting.  

I also fixed [your other request](https://github.com/quasarframework/quasar/issues/3482).

I'm sure this won't be taken as I've done it, as I'm not certain of everything in play, but it's a start. :)

Scott
